miniapps/solvers/darcy_solver.hpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2010-2023, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_DARCY_SOLVER_HPP
+#define MFEM_DARCY_SOLVER_HPP
+
+#include "mfem.hpp"
+#include <memory>
+#include <vector>
+
+namespace mfem
+{
+namespace blocksolvers
+{
+struct IterSolveParameters
+{
+   int print_level = 0;
+   int max_iter = 500;
+   double abs_tol = 1e-12;
+   double rel_tol = 1e-9;
+};
+
+void SetOptions(IterativeSolver& solver, const IterSolveParameters& param);
+
+/// Abstract solver class for Darcy's flow
+class DarcySolver : public Solver
+{
+protected:
+   Array<int> offsets_;
+public:
+   DarcySolver(int size0, int size1) : Solver(size0 + size1), offsets_(3)
+   { offsets_[0] = 0; offsets_[1] = size0; offsets_[2] = height; }
```

Comment:
You could initialize `offsets_` in a "fancier" way like this:
```suggestion
   DarcySolver(int size0, int size1) : Solver(size0 + size1), offsets_({0, size0, height}) { }
```